File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;

        int count = 0;

        for(int i = 1; i < n; i++) {
            if(nums[i-1] > nums[i]) count++;
        }

        if(nums[n-1] > nums[0]) count++;

        return count <= 1;
    }
}



// a a+1 a+3 a+3 . .... a+10


// 10 12 13 13 1 2 15

File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int stepDownCount = 0;
        int n =nums.size();
        for(int i=1;i<nums.size();i++)
        {
            if(nums[i]<nums[i-1])
            {
                stepDownCount+=1;
            }
        }
        if(stepDownCount==0)
        {
            return true;
        }
        else if(stepDownCount==1)
        {
            return nums[0]>=nums[n-1];
        }
        return false;
    }
};